src/transports/smart_pkt.c

```diff
@@ -291,11 +291,13 @@ static int ng_pkt(git_pkt **out, const char *line, size_t len)
 	pkt->ref = NULL;
 	pkt->type = GIT_PKT_NG;
 
+	eol = line + len;
+
 	if (len < 3)
 		goto out_err;
 	line += 3; /* skip "ng " */
-	len -= 3;
-	if (!(ptr = memchr(line, ' ', len)))
+
+	if (!(ptr = memchr(line, ' ', eol - line)))
 		goto out_err;
 	len = ptr - line;
```

Comment:
This line is the problem. We overwrite `len` with the length of the reference and thus loose track of the overall line length. Thus the following computations on `len` are invalid when we try to see how much characters are left in the pkt-line.